### ISSUE TITLE
Support Aeson 2.2

### ISSUE THREAD

**Issue by Vekhir** [hslua/hslua#139]
Hi!
When running the tests for `hslua-aeson` v2.3.0, I get the following (or similar) error:
```
Running 1 test suites...
Test suite test-hslua-aeson: RUNNING...
hslua-aeson
  Value
    can be round-tripped through the stack:              FAIL
      *** Failed! Assertion failed (after 60 tests and 8 shrinks):
      Number 7.0e22
      Use --quickcheck-replay=632700 to reproduce.
      Use -p '/can be round-tripped through the stack/' to rerun this test only.
    can roundtrip a bool nested in 50 layers of arrays:  OK
      +++ OK, passed 100 tests.
    can roundtrip a bool nested in 50 layers of objects: OK
      +++ OK, passed 100 tests.
    can roundtrip a null nested in 50 layers of objects: OK
      +++ OK, passed 100 tests.
  via JSON
    can roundtrip 'Maybe Text' via JSON:                 OK
      +++ OK, passed 100 tests.
    can roundtrip '(Int, Float)' via JSON:               OK
      +++ OK, passed 100 tests.
    can roundtrip 'Either Bool Text' via JSON:           OK
      +++ OK, passed 100 tests.
  special encodings
    __toaeson
      respect __toaeson metamethod:                      OK
    __tojson
      respect __tojson metamethod:                       OK

1 out of 9 tests failed (0.02s)
Test suite test-hslua-aeson: FAIL
Test suite logged to: dist/test/hslua-aeson-2.3.0-test-hslua-aeson.log
0 of 1 test suites (0 of 1 test cases) passed.

```
This occurs using the [official Arch package](https://archlinux.org/packages/extra/x86_64/haskell-hslua-aeson/) and compiling with GHC 9.0.2, but I have updated many Haskell dependencies, including Aeson to 2.2 (which I had to manually ignore). I don't think that's necessarily the issue, though.

Can you reproduce this issue? I can provide more infos to my setup if requested.

## System information
OS: Arch Linux
Kernel: Linux 6.4.12-arch1-1
GHC: 9.0.2
hslua-aeson: 2.3.0
aeson: 2.2
hslua-marshalling: 2.3.0

Thank you for your time,
Vekhir

**Comment by tarleb** [comment]
Thanks for the report. I'm not entirely sure what's going on yet, but my theory for now is that newer Aeson version differ in the range of values produced for QuickCheck: not all JSON numbers can be converted to Lua numbers without loss of precision, I think, and the new Aeson might produce those. There's probably not much we can reasonably do outside of fixing the tests.

Do these test failures block you in one way or the other, i.e., should we do a release after fixing the tests?

**Comment by Vekhir** [comment]
Please do a release with the fix (and bumped aeson) when you are done, it does block the upgrade to Aeson 2.2 in the Arch repos.

**Comment by tarleb** [comment]
OK, I'm updating the issue title accordingly.

**Comment by peti** [comment]
Is there any chance we could get support for aeson-2.2.x soon'ish? As it is, hslua-aeson blocks a lot of updates in the openSUSE distribution because it locks us down to the old aeson version.

**Comment by tarleb** [comment]
Apologies for the wait. This is now fixed in hslua-aeson-2.3.1.
